### ISSUE TITLE
Setup SMS notifications with SNS

### ISSUE THREAD

**Issue by QaysarA** [department-of-veterans-affairs/notification-api#236]
# Value Statement
**As a** VANotify customer
**I want to** send text messages 
**So that** text message notifications can be sent to the Veteran 

# Acceptance Criteria
**GIVEN** VANotify receives a request to send a text notification with `phone_number` specified
**WHEN** SNS is setup for text notifications
**THEN** a text is sent to the respective phone number.

**GIVEN** SNS is setup for text notifications
**WHEN** a text is sent
**THEN** SNS is the only SMS provider enabled.

# Checklist
- [ ] Add `SNS_SMS_CLIENT_ENABLED` feature flag to `PROVIDER_FEATURE_FLAGS` dict
    - Should this be toggled in env variables or in `config.py` like `GOVDELIVERY_EMAIL_CLIENT_ENABLED`?
- [x] Make SNS the primary sms provider by setting it to a higher priority (lower number) through a db migration - similar to `0307a_update_twilio_priority`. May need to lower Twilio's priority since we are not using it.
    - SNS is already an SMS provider - see migration `0297a_add_sns_provider`
    - Current SMS priorities: Twilio (5), SNS (10), MMG (15), Firetext (20), Loadtesting (30)
- [x] Check that SNS is enabled as provider
    - It is already "active" in the DB, so this may just be testing that when we try to send a text, it goes to use SNS first rather than Twilio.
- [x] Refactor `aws_sns.py` tests to not rely on the SNS client being enabled 
    - can use the test changes from this PR: https://github.com/department-of-veterans-affairs/notification-api/pull/149/files
    - however, we want to **keep** the initialization of AwsSnsClient (in `app/__init__.py`)

- [x] Execution role may need SNS permissions (terraform)

- Create an organization for our team (VANotify), testing service, template for text message -> add info for organization/service for testing in vanotify-team; service key in param store

- [ ] Add user flow test for sending text
- [x] Updates to Postman collection, if needed
- [x] Research mode is working with SNS

# Assumptions
- SNS is already in the codebase. There should be few actual code changes to send a text.
- One text message
- Sending a request with `recipient_identifier` should work as expected with text message (no changes should need to be made). Instead of the text sending, it should go to one of the `lookup_contact_information` queues

# Additional Info/Resources
- Sending transactional texts with SNS - just sending a message, don't care about from_number or delivery status
- Filip is disabling provider switching errors that are happening

# Out of Scope
- Setup of short code or long code -> these are created by a support ticket to AWS, which we are waiting on
- Updating the status is not in scope for this card
- Pilot VEText use case with 1 clinic with long code (specific area code)

# Open Questions
- Do we want to remove some of the providers we are not using? Or should we keep them in case they are used in the future?

**Comment by ffafara-tw** [comment]
Regarding:

> Make SNS the primary sms provider by setting it to a higher priority (lower number) through a db migration - similar to 0307a_update_twilio_priority. May need to lower Twilio's priority since we are not using it.
> 
> SNS is already an SMS provider - see migration 0297a_add_sns_provider
> Current SMS priorities: Twilio (5), SNS (10), MMG (15), Firetext (20), Loadtesting (30)

I have adjusted priority based using API requests, since SNS will not be the default provider in the future.
I have also deactivated unused providers.

**Comment by ffafara-tw** [comment]
Regarding:

> Add SNS_SMS_CLIENT_ENABLED feature flag to PROVIDER_FEATURE_FLAGS dict

will discuss if this is necessary during dev sync

**Comment by jsmithVA** [comment]
Reviewed in desk check and a test review with Filip. Passing to Sign off.
